pkg/providers/instancetype/suite_test.go

```diff
@@ -1716,6 +1718,159 @@ var _ = Describe("InstanceTypes", func() {
 			})
 		})
 	})
+	Context("Provider Cache", func() {
+		It("changes to kubelet configuration fields should result in a different set of instances types", func() {
+			// We should expect these kubelet configuration fields to change the result of the instance type call
+			// kubelet.kubeReserved
+			// kubelet.systemReserved
+			// kubelet.evictionHard
+			// kubelet.evictionSoft
+			// kubelet.maxPods
+			nodePool.Spec.Template.Spec.Kubelet = &corev1beta1.KubeletConfiguration{
+				KubeReserved:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("1")},
+				SystemReserved: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1")},
+				EvictionHard:   map[string]string{"memory.available": "5%"},
+				EvictionSoft:   map[string]string{"nodefs.available": "10%"},
+				MaxPods:        aws.Int32(10),
+			}
+			kubeletChanges := []*corev1beta1.KubeletConfiguration{
```

Comment:
Does it make sense to make this a `DescribeTable` so each test is sectioned out rather than a single test that iterates through each thing?

Reply:
I opted to not make it a DescribeTable call to guarantee that the `awsEnv` cache for instanceTypes is not cleared out in between test runs

Reply:
Realized I was BeforeEach` is not run on `DescribeTable` https://github.com/onsi/ginkgo/issues/378

Reply:
We will need to generate all the different configuration and compare the list of instancetypes that are contained in the cache, so for that reason they will need to be in the same `IT` block.

Reply:
Seems to me like we can make this a table. Each iteration of the table is a different application of the kubeletChanges. It would change the For loop on L1753

Reply:
Yeah, but then we need to make sure that each set of instance types is compared against each other in the For loop on L1773.

Reply:
Since it's non-obvious why we did this in the first place, should we consider adding a comment that explains why we are testing these things in a single It() block so that it's clearer to the reader?